### ISSUE TITLE
Auth module is swallowing IDBRequest errors

### ISSUE THREAD

**Issue by piuccio** [firebase/firebase-js-sdk#1397]
### [REQUIRED] Describe your environment

  * Operating System version: Android version 7.0 and 7.1
  * Browser version: Chrome 67 ~ 70
  * Firebase SDK version: 5.5.0 ~ 5.5.9
  * Firebase Product: auth (possibly others)

#### Steps to reproduce:

I have no idea on how to reproduce, but every once in a while I have some errors in Sentry, so far they're all from Chrome on Android with the following stack trace

```
{
message: , 
name: Error, 
stack: 
Error
    at IDBRequest.e.onerror (https://my.site.com/vendors~main.chunk.js:42:261322)
}
```

Not very descriptive.

The error happens soon after I try to log in user with a custom token

```js
await firebase.auth().signInWithCustomToken(token);
```

I see these two POST requests

POST https://www.googleapis.com/identitytoolkit/v3/relyingparty/verifyCustomToken?key=key [200]
POST https://www.googleapis.com/identitytoolkit/v3/relyingparty/getAccountInfo?key=key [200]

And then the error

#### Relevant Code:

The exception comes from this line https://github.com/firebase/firebase-js-sdk/blob/d2f0667fceb4bb2cbc8b08f536bf3785d1de9801/packages/auth/src/storage/indexeddb.js#L402 

```javascript
request.onerror = function(event) {
  reject(new Error(event.target.errorCode));
};
```

and the only reference I could find of `errorCode` is from [this MDN article](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API/Using_IndexedDB#Handling_Errors)

I know that MDN is generally reliable, but I was confused because in my logs that value is `undefined` and if you look at [this page (IDBRequest.onerror)](https://developer.mozilla.org/en-US/docs/Web/API/IDBRequest/onerror) instead, there's no mention of `errorCode`.

I tried to find an answer in the w3c spec and `errorCode` is only referenced until [this version of the spec](https://www.w3.org/TR/2011/WD-IndexedDB-20110419/) dated April 2011. Already the version from December 2011 doesn't have any reference of it.

The latest specification for IDB2 - IDBRequest can be [found here](https://www.w3.org/TR/IndexedDB/#request-api) and if I understand it correctly that code should be rewritten as

```diff
request.onerror = function(event) {
--  reject(new Error(event.target.errorCode));
++  reject(request.error);
};
```

That's because `request.error` is of type `DOMException` so it's already descriptive. Alternatively, to enforce the error to be `instanceof Error` you could do

```js
reject(new Error(request.error.name));
```

I would make a PR myself but I couldn't find a way to fire IndexDB errors and verify my assumptions. Also that code `event.target.errorCode` is used in other modules, so ideally it should change elsewhere, it seems too big of a task for someone that has no idea on how this project works.

**Comment by jsayol** [comment]
This does seem like an mistake. `event.target` is the `IDBRequest` object and it doesn't have an `errorCode` property. It does have `error` though, which is a `DOMException`. That line should probably be:

```js
reject(new Error(event.target.error.name));
```

Or even just
```js
reject(event.target.error);
```

since that would contain both `name` and `message`.

A quick example showing all this:
```js
const dbOpenRequest = window.indexedDB.open("toDoList", 1);

dbOpenRequest.onupgradeneeded = function(event) {
  if (event.oldVersion < 1) {
    dbOpenRequest.result.createObjectStore("toDoList");
  }
}

dbOpenRequest.onsuccess = function(event) {
  const db = dbOpenRequest.result;
  const transaction = db.transaction(["toDoList"], "readwrite");
  const objectStore = transaction.objectStore("toDoList");

  const key = Date.now();
  const objectStoreRequest1 = objectStore.add('hello', key);
  const objectStoreRequest2 = objectStore.add('duplicate key!', key);

  objectStoreRequest2.onerror = function(event) {
    console.log({
      target: event.target.constructor.name,
      'target.error': event.target.error.constructor.name,
      'target.error.message': event.target.error.message,
      'target.error.name': event.target.error.name,
      'target.errorCode': typeof event.target.errorCode,
    });
  };
};
```
Which gives:
```js
{
  target: "IDBRequest"
  target.error: "DOMException"
  target.error.message: "Key already exists in the object store."
  target.error.name: "ConstraintError"
  target.errorCode: "undefined"
}

**Comment by bojeil-google** [comment]
@jsayol's solution looks good. Will send PR to fix this tomorrow. Thanks.

**Comment by jsayol** [comment]
FYI this can be closed, since the fix was published with 5.6.0.
